packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -544,11 +544,15 @@ contract Bridge is EssentialContract, IBridge {
         view
         returns (bool)
     {
-        return signalService.proveSignalReceived({
+        try signalService.proveSignalReceived({
```

Comment:
Common misconception with solidity which is coming from the syntax of `try ... catch` is that it will catches every reverts, but it will not, by default catches custom error inside the expression.

See a good explanation here:
https://forum.soliditylang.org/t/call-for-feedback-the-future-of-try-catch-in-solidity/1497

If we want to somehow propoerly try-catch custom errors, here is a pseudo:
```
// A low level call to the contract
 (bool success, bytes memory returnData) =
            address(signalService).call( 
                abi.encodePacked(
                    signalService.proveSignalReceived.selector,
                    abi.encode(srcChainId, app,signal,proof)
                )
            );
        if (success) { 
            // proveSignalReceived completed successfully (did not revert)
            return true;
        } else { 
            // proveSignalReceived reverted.
            return false;
        }
```

Reply:
I would say in our case try/catch would be the one to use though, unless I misunderstand something. We're only interested in catching errors in the called function, and the additional checks on the smart contract existing before the call should revert because it means that the smart contract is not setup correctly, so still reverting on those would be the expected behavior.

Of course in the end they are almost the same, and should be fine to go with both.

Reply:
> Of course in the end they are almost the same, and should be fine to go with both.

You might (most prob.) right here, but i still prefer a solution we already 'know' from `_invokeMessage`. The reason i dig up that article is, because i know at one point in time we struggled with `try..catch` (maybe was misused (?)) but we even removed it.

Anyways as it is ambigous - hence diff. behavior in other languages - i still prefer this one.